### ISSUE TITLE
Fix id in the appdata file

### ISSUE THREAD

**Issue by dagargo** [dagargo/elektroid#163]
ID in the `appdata` file does not follow the specification.

https://github.com/dagargo/elektroid/blob/a8162d52511812804b1a4949f01da529ef56c134/res/elektroid.appdata.xml#L4

See https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#id and https://www.freedesktop.org/software/appstream/docs/chap-Validation.html#asv-cid-desktopapp-is-not-rdns. This might break things to the point of resubmitting the application to Flathub so this needs to be checked.

Besides, changing the ID would need to change the `desktop` file but this makes the application icon to not show. Perhaps the icon names need to be changed too.

Finally, it needs to be checked if this ID would break the already existing packages.

**Comment by dagargo** [comment]
@snd1, regarding the Debian package, would changing the `id` from "elektroid.desktop" to "io.github.dagargo.Elektroid" in this file break anything?

https://github.com/dagargo/elektroid/blob/a8162d52511812804b1a4949f01da529ef56c134/res/elektroid.appdata.xml#L4

As icon names would need to be changed too, would these renamings be an issue?

**Comment by snd1** [comment]
@dagargo Sorry for the late reply. Changing the name is ok, but the file should be renamed to "io.github.dagargo.elektroid.desktop" so with the desktop file name, otherwise it will not work afaik

**Comment by snd1** [comment]
To clarify:

<id>io.github.dagargo.elektroid</id>

and

<launchable type="desktop-id">io.github.dagargo.elektroid.desktop</launchable>

**Comment by dagargo** [comment]
Thanks for the explanation.

Does this affect the Debian package for the worse? I don't wanna make things more complicated.

**Comment by snd1** [comment]
No, it should not affect the Debian Package, the new package will replace the old desktop file then.

**Comment by dagargo** [comment]
ID changed in 89cd5bbff05473d6ab614787f83eecc20e569cd6.
Some other files needed renaming as well and some references to these were updated.

**Comment by snd1** [comment]
Yes, the soft freeze begin on 2025-04-15: https://release.debian.org/testing/freeze_policy.html
And yes sure it would be cool to release 3.2 before 👍 But take your time, a backport is possible later too :)

**Comment by dagargo** [comment]
Thanks for the confirmation, @snd1.

The idea is to release 3.2 by during the first week of April and I'll try to implement #172 for then.
